WordPress/gutenberg
# Switching pattern categories inserter to Tabs component with arrow key navigation

Fixes https://github.com/WordPress/gutenberg/issues/60209

## What?

Improves the accessibility and keyboard navigation of using the pattern categories list in the inserter.

## Why?

The current implementation is cumbersome with too many tabstops and the panel is disconnected from the activating button, making it hard to switch between categories using a keyboard.

## How?

Changes the structure of the pattern categories from a listbox of buttons that open a panel to Tabs implementation that considers all the categories as one tabstop with arrow keys to move between the categories.


### Testing Instructions for Keyboard

- Open Block inserter (in post editor and in site editor)
- Select Pattern Tab
- Tab to patten categories panel
- Use arrows to navigate between the patterns
- Select a pattern
- Tab into patterns panel
- Shift + Tab out
- Focus should be on the selected category
- Arrow between patterns and repeat switching between categories


## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/967608/e06aa513-cd78-4722-8172-239fa126e849
